dotCMS/src/main/java/com/dotcms/http/CircuitBreakerUrl.java

```diff
@@ -83,18 +84,16 @@ public class CircuitBreakerUrl {
     private int response = -1;
     private Header[] responseHeaders;
     private final boolean allowRedirects;
-    private final boolean throwWhenNot2xx;
+    private final boolean throwWhenError;
+    private final Function<Integer, HttpStatusCodeException> overrideException;
```

Comment:
Adding a couple of exception handling options that remain turned off for backward compatibility.